src/prefect/cli/deployment.py

```diff
@@ -267,6 +272,21 @@ async def apply(
         except Exception as exc:
             exit_with_error(f"'{path!s}' did not conform to deployment spec: {exc!r}")
 
+        if upload:
+            file_count = await deployment.upload_to_storage()
+            if file_count and deployment.storage:
+                location = (
```

Comment:
Is the `location` something the `Deployment` class should know rather than building it in the CLI?

Reply:
I could expose it as a computed attribute for sure, and I think that would help users discover how entrypoint, path, and storage are all related.

Would you be OK if I did that in a follow-up PR?

Reply:
Actually if I'm going to write some more tests I can do it here :+1: